core/src/test/java/org/apache/iceberg/expressions/TestInclusiveMetricsEvaluatorWithExtract.java

```diff
@@ -0,0 +1,687 @@
+        .as("Should skip: greaterThanOrEqual on all null column")
+        .isFalse();
+
+    assertThat(shouldRead(equal(extract("all_nulls", "$.event_id", "long"), 30)))
+        .as("Should skip: equal on all null column")
+        .isFalse();
+
+    assertThat(shouldRead(notEqual(extract("all_nulls", "$.event_id", "long"), 30)))
+        .as("Should read: notEqual on all null column")
+        .isTrue();
+
+    assertThat(shouldRead(in(extract("all_nulls", "$.event_type", "string"), "abc", "def")))
+        .as("Should skip: in on all nulls column")
+        .isFalse();
+
+    assertThat(shouldRead(notIn(extract("all_nulls", "$.event_type", "string"), "abc", "def")))
+        .as("Should read: notIn on all nulls column")
+        .isTrue();
+
+    assertThat(shouldRead(startsWith(extract("all_nulls", "$.event_type", "string"), "a")))
+        .as("Should skip: startsWith on all null column")
+        .isFalse();
+
+    assertThat(shouldRead(notStartsWith(extract("all_nulls", "$.event_type", "string"), "a")))
+        .as("Should read: notStartsWith on all null column")
+        .isTrue();
+
+    assertThat(shouldRead(isNaN(extract("all_nulls", "$.measurement", "double"))))
+        .as("Should skip: no NaN value in all null column")
+        .isFalse();
+
+    assertThat(shouldRead(notNaN(extract("all_nulls", "$.measurement", "double"))))
+        .as("Should read: all null column has non-NaN values")
+        .isTrue();
+  }
+
+  @Test
+  public void testIsNaNAndNotNaN() {
+    assertThat(shouldRead(isNaN(extract("variant", "$.measurement", "double"))))
+        .as("Should read: variant may contain NaN")
+        .isTrue();
+
+    assertThat(shouldRead(notNaN(extract("variant", "$.measurement", "double"))))
+        .as("Should read: variant may contain non-NaN")
+        .isTrue();
+  }
+
+  @Test
+  public void testMissingColumn() {
+    assertThatThrownBy(() -> shouldRead(lessThan(extract("missing", "$", "int"), 5)))
+        .isInstanceOf(ValidationException.class)
+        .hasMessageContaining("Cannot find field 'missing'");
+  }
+
+  private static final Expression[] MISSING_STATS_EXPRESSIONS =
+      new Expression[] {
+        lessThan(extract("variant", "$.missing", "long"), 5),
+        lessThanOrEqual(extract("variant", "$.missing", "long"), 30),
+        equal(extract("variant", "$.missing", "long"), 70),
+        greaterThan(extract("variant", "$.missing", "long"), 78),
+        greaterThanOrEqual(extract("variant", "$.missing", "long"), 90),
+        notEqual(extract("variant", "$.missing", "long"), 101),
+        isNull(extract("variant", "$.missing", "string")),
+        notNull(extract("variant", "$.missing", "string")),
+        isNaN(extract("variant", "$.missing", "float")),
+        notNaN(extract("variant", "$.missing", "float"))
+      };
+
+  @ParameterizedTest
+  @FieldSource("MISSING_STATS_EXPRESSIONS")
+  public void testMissingStats(Expression expr) {
```

Comment:
If the bounds are in metadata, they are trusted. Bounds should be missing in cases where there are variant values with incompatible types.